julia/core.py

```diff
@@ -443,26 +496,44 @@ def __init__(self, init_julia=True, jl_runtime_path=None, jl_init_path=None,
                 # configuration and so do any packages that depend on it.
                 self._call(u"unshift!(Base.LOAD_CACHE_PATH, abspath(Pkg.Dir._pkgroot()," +
                     "\"lib\", \"pyjulia%s-v$(VERSION.major).$(VERSION.minor)\"))" % sys.version_info[0])
-                # If PyCall.ji does not exist, create an empty file to force
-                # recompilation
+
+                # If PyCall.jl is already pre-compiled, for the global
+                # environment, hide it while we are loading PyCall.jl
+                # for PyJulia which has to compile a new cache if it
+                # does not exist.  However, Julia does not compile a
+                # new cache if it exists in Base.LOAD_CACHE_PATH[2:end].
+                # https://github.com/JuliaPy/pyjulia/issues/92#issuecomment-289303684
                 self._call(u"""
-                    isdir(Base.LOAD_CACHE_PATH[1]) ||
-                        mkpath(Base.LOAD_CACHE_PATH[1])
-                    depsfile = joinpath(Base.LOAD_CACHE_PATH[1],"PyCall.ji")
-                    isfile(depsfile) || touch(depsfile)
+                for path in Base.LOAD_CACHE_PATH[2:end]
+                    cache = joinpath(path, "PyCall.ji")
+                    backup = joinpath(path, "PyCall.ji.backup")
+                    if isfile(cache)
+                        mv(cache, backup; remove_destination=true)
```

Comment:
This is a bit orthogonal to this PR but I figured fixing #175 is required for passing the test.  Merging this fix should also resolve #109 and #169.

In `.travis.yml` we have:

https://github.com/JuliaPy/pyjulia/blob/138d47dce6d6189be9ab6a8ea330cec3a237a8be/.travis.yml#L62-L64

Previously, `use_separate_cache` was `True` for both `py` and `py27` virtual environments.  Thus, no global cache `~/.julia/lib/v0.6/PyCall.ji` existed while running the above test.  However, with this patch, `use_separate_cache` is `False` in `py` because the code I added sees that `PYTHON` configured for `PyCall` uses the same `libpython` used by `py` virtual environment prepared by `tox`.  So, after running `tox -e py`, there is a file `~/.julia/lib/v0.6/PyCall.ji`.  Now then running `tox -e py27` hits the bug #175 because `julia` 0.6 does not recompile `PyCall` when it finds `~/.julia/lib/v0.6/PyCall.ji` in `Base.LOAD_CACHE_PATH[2]`.